contracts/dao/PancakeStaking.sol

```diff
@@ -188,4 +227,25 @@ contract PancakeStaking is OwnableUpgradeable, ReentrancyGuardUpgradeable {
     function setHarvestTimeGap(uint256 _harvestTimeGap) external onlyOwner {
         harvestTimeGap = _harvestTimeGap;
     }
+
+    /**
+      * @dev set emergency mode
+      * @param _emergencyMode emergency mode
+      */
+    function setEmergencyMode(bool _emergencyMode) external onlyOwner {
+        emergencyMode = _emergencyMode;
+        emit SetEmergencyMode(_emergencyMode);
+    }
+
+    function emergencyWithdraw(address pool) external onlyOwner {
```

Comment:
The emergency mode is an irreversible action, enable it in `emergencyWithdraw`

Reply:
done in 49ec5670cb578e687c2bd3d9cd4fabf31f7d2863